src/utils/hooks/useMediaQuery.ts

```diff
@@ -0,0 +1,32 @@
+import { useEffect, useState } from 'react';
+
+const MOBILE_QUERY = '(max-width: 576px)';
+
+export default function useMediaQuery(query = MOBILE_QUERY): boolean {
+  const getMatches = (matchedQuery: string): boolean => {
+    // Prevents SSR issues
+    if (typeof window !== 'undefined') {
+      return window.matchMedia(matchedQuery).matches;
+    }
+    return false;
+  };
+  const [matches, setMatches] = useState(getMatches(query));
```

Comment:
정말 사소한 최적화이지만, `matches` 상태가 변경되었을 때 초기값을 사용하지 않음에도 `getMatches` 가 호출될것 같아요. 익명 함수로 감싸주는 것이 어떨까요?

```suggestion
  const [matches, setMatches] = useState(() => getMatches(query));
```

Reply:
오! state 초기화엔 함수를 사용! 블로그로 배웠던거를 여기서 보네요 ㅋㅋㅋ